source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -0,0 +1,170 @@
+#include "extensions/filters/http/cache/cache_headers_utils.h"
+
+#include <array>
+#include <string>
+
+#include "envoy/common/time.h"
+
+#include "absl/algorithm/container.h"
+#include "absl/strings/ascii.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+// Utility functions used in RequestCacheControl & ResponseCacheControl
+namespace {
+OptionalDuration parseDuration(absl::string_view s) {
+  OptionalDuration duration;
+  // Strip quotation marks if any
+  if (s.size() > 1 && s.front() == '"' && s.back() == '"') {
+    s = s.substr(1, s.size() - 2);
+  }
+  long num;
+  if (absl::SimpleAtoi(s, &num) && num >= 0) {
+    // s is a valid string of digits representing a positive number
+    duration = std::chrono::seconds(num);
+  }
+  return duration;
+}
+
+inline std::pair<absl::string_view, absl::string_view>
+separateDirectiveAndArgument(absl::string_view full_directive) {
+  return absl::StrSplit(absl::StripAsciiWhitespace(full_directive), absl::MaxSplits('=', 1));
+}
+} // namespace
+
+// The grammar for This Cache-Control header value should be:
+// Cache-Control   = 1#cache-directive
+// cache-directive = token [ "=" ( token / quoted-string ) ]
+// token           = 1*tchar
+// tchar           = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+"
+//                 / "-" / "." / "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+// quoted-string   = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+// qdtext          = HTAB / SP /%x21 / %x23-5B / %x5D-7E / obs-text
+// obs-text        = %x80-FF
+// quoted-pair     = "\" ( HTAB / SP / VCHAR / obs-text )
+// VCHAR           =  %x21-7E  ; visible (printing) characters
+
+RequestCacheControl::RequestCacheControl(absl::string_view cache_control_header) {
+  must_validate_ = no_store_ = no_transform_ = only_if_cached_ = false;
```

Comment:
init these in the declaration.

Reply:
Member initialization of bitfields in declarations is a C++20 feature.

Reply:
If these are replaced by booleans I will initialize them in the declaration